### PR TITLE
feature: add support for binary files

### PR DESCRIPTION
**Pull request by pixelastic**
As discussed in #30, I added the logic to add binary files through their blob shas.

If the changed file value is a string, it uses the previous implementation. If not, it will first create the blob, and then use its sha value in the newly created tree.

I did some small refactoring for readability on my way: creating a `isString` method and replacing `changes.files[path]` with `value` in the loop. 

I also updated the example in the documentation, and removed the item from the TODO. Running `npm run lint:fix` reformatted the code markdown.

Tests are utterly failing though:
```
Suites:   6 failed, 1 passed, 7 of 7 completed
Asserts:  249 failed, 156 passed, of 405
```

I'd love to have a look and fix them as well as add the missing tests for coverage, but I have no experience with `tap` and couldn't find how to run only one test at a time. I tested it on my own repo, and it works.

Let me know what you think, and how I could fix the tests and I'll give it a second pass. Thanks.

### CODE REVIEW

File: lib/create-pull-request.js [hunk 1]
```diff
@@ -80,10 +87,29 @@ async function createPullRequest(
           }
         }
 
+        // Text files can be changed through the .content key
+        if (isString(value)) {
```
Comment (gr2m) [inline lib/create-pull-request.js:91]: Can you elaborate why you don't check with `typeof value === 'string'`?

Comment (gr2m) [reply lib/create-pull-request.js:91]: All existing tests pass when checking with `typeof value === 'string'` instead

**Comment by gr2m** [comment]
@pixelastic I've added tests, simplified the string check, and amended the contributor docs. Let me know what you think

File: lib/create-pull-request.js (hunk 1 above)
Comment (pixelastic) [reply lib/create-pull-request.js]: TBH I never know how to test for sure if a var is a string in plain JS (I usually have _.isString from lodash). So I went [to Stack Overflow](https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string-in-javascript) and this seemed like the most robust answer but I didn't dig any deeper than that...

So, in a nutshell, I'm fine with the `typeof` check

**Comment by pixelastic** [comment]
Looks perfect; thanks for cleaning it up!

**Pull request merged**